language/stdlib/modules/account_limits.move

```diff
@@ -0,0 +1,240 @@
+        window_outflow: u64,
+        // The inflow during this time window
+        window_inflow: u64,
+        // The balance that this account has held during this time period.
+        tracked_balance: u64,
+    }
+
+    // Return back a capability to allow calling the public mutation
+    // functions in this module. The caller must be the singleton_addr for
+    // the AccountTrack module.
+    public fun grant_account_tracking(): MutationCapability {
+        // This address needs to match the singleton_addr in AccountTrack
+        Transaction::assert(Transaction::sender() == 0xA550C18, 2);
+        MutationCapability{}
+    }
+
+    // Determine if the depositing of `amount` of `CoinType` coins into an
+    // account with `receiving_window_info` is amenable with their limits.
+    // Returns false if this violates the limits. Mutative.
+    public fun update_deposit_limits<CoinType>(
+        amount: u64,
+        receiving_limits_addr: address,
+        receiving_window_info: &mut Window,
+        _cap: &MutationCapability
+    ): bool acquires LimitsDefinition {
+        can_receive<CoinType>(
+            amount,
+            receiving_window_info,
+            borrow_global<LimitsDefinition>(receiving_limits_addr),
+        )
+    }
+
+    // Determine if withdrawing `amount` of `CoinType` coins from
+    // the account with `account_window_info` would violate the
+    // LimitsDefinition held at the `limits_addr`. Returns false if this is
+    // not permissible. Mutative.
+    public fun update_withdrawal_limits<CoinType>(
+        amount: u64,
+        limits_addr: address,
+        account_window_info: &mut Window,
+        _cap: &MutationCapability
+    ): bool acquires LimitsDefinition {
+        can_withdraw<CoinType>(
+            amount,
+            account_window_info,
+            borrow_global<LimitsDefinition>(limits_addr),
+        )
+    }
+
+    // All unhosted accounts will have this published in their account
+    // info. Root accounts for multi-account entities will hold this
+    // resource in their account information.
+    public fun create(): Window {
+        Window {
+            window_start: current_time(),
+            window_outflow: 0,
+            window_inflow: 0,
+            tracked_balance: 0,
+        }
+    }
+
+    // Anyone can publish a LimitsDefinition resource under their address. But
+    // it does nothing until the association certifies the LimitsDefinition.
+    public fun publish_limits_definition(
+        max_outflow: u64,
+        max_inflow: u64,
+        max_holding: u64,
+        time_period: u64
+    ) {
+        move_to_sender(LimitsDefinition {
+            max_outflow,
+            max_inflow,
+            max_holding,
+            time_period,
+            is_certified: false,
+        });
+    }
+
+    // Unrestricted accounts are represented by setting all fields in the
+    // limits definition to 0.
+    public fun publish_unrestricted_limits() {
```

Comment:
Should this be permissioned?

Reply:
Nope! This is fine -- the `limits_definition` that is published with this is  uncertified so doesn't change anything.